src/targets/gpu/compile_gen.cpp

```diff
@@ -169,6 +172,116 @@ bool preload::is_preloading() const
     return std::accumulate(args.begin(), args.end(), false, std::logical_or<>{});
 }
 
+static std::size_t integer_divide_ceil(std::size_t x, std::size_t y)
+{
+    return (x + y - std::size_t{1}) / y;
+}
+
+static std::size_t compute_tile_factor(std::size_t r, std::size_t max_size = 64)
+{
+    std::size_t n = 1;
+    auto factors  = make_array(2, 3, 5, 7, 11);
+    while(n < max_size)
+    {
+        // NOLINTNEXTLINE(readability-qualified-auto)
+        auto it = std::find_if(factors.begin(), factors.end(), [&](auto d) { return r % d == 0; });
+        if(it == factors.end())
+            break;
+        r /= *it;
+        n *= *it;
+    }
+    return n;
+}
+
+tile tile::elements(const std::vector<shape>& inputs, std::size_t noutputs)
+{
+    tile result;
+    auto ndim = inputs.front().ndim();
+    std::vector<std::size_t> faxes;
+    std::transform(
+        inputs.begin(), inputs.end(), std::back_inserter(faxes), MIGRAPHX_LIFT(find_fast_axis));
+    result.axis = std::accumulate(faxes.begin(), faxes.end(), ndim, MIGRAPHX_LIFT(std::min));
+    if(result.axis >= (ndim - 1))
+        return {};
+    auto select = [&](auto m) {
+        return [&, m](std::size_t faxis, shape input) {
+            if(input.broadcasted())
+                return none;
+            if(faxis < (ndim - 1))
+                return m;
+            return none;
+        };
+    };
+    std::transform(faxes.begin(),
+                   faxes.end() - noutputs,
+                   inputs.begin(),
+                   std::back_inserter(result.args),
+                   select(load));
+    std::transform(faxes.end() - noutputs,
+                   faxes.end(),
+                   inputs.end() - noutputs,
+                   std::back_inserter(result.args),
+                   select(store));
+
+    auto nargs = std::count_if(
+        result.args.begin(), result.args.end(), [](auto m) { return m != mode::none; });
+    // TODO: Handle tiling more than one arguments
+    if(nargs != 1)
+        return {};
+
+    const auto& s = inputs.front();
+    auto dim1     = compute_tile_factor(s.lens()[result.axis]);
+    auto dim2     = compute_tile_factor(s.lens().back(), 4096 / dim1);
+    if(dim1 == 1 or dim2 == 1)
+        return {};
```

Comment:
If I'm understanding this correctly, this code won't tile a larger prime number dimension like 1009. It's trying to find a tile that fits the tensor without going out of bounds. Can we also support these prime number dimensions by having a bounds check in the kernel?

Reply:
I dont think we want a larger prime. My understanding was the use of primes was related to having the smallest possible irreducible kernel size and going above 13 wasn't used/ we never ran into a case for the higher primes.

Reply:
Not about larger primes for the factorization, I mean if we have a larger prime number for the dimension to tile. Since we're trying to use the largest tile size that fits on LDS?

Reply:
> Can we also support these prime number dimensions by having a bounds check in the kernel?

We would need to do padding, which will be a little more complicated, but its definitely something we should probably look into the future.

Reply:
Issue to track https://github.com/ROCm/AMDMIGraphX/issues/3427